starter_scripts/exaslct_within_docker_container.sh

```diff
@@ -11,12 +16,95 @@ else
+relevant_mount_point_arguments["--output-directory"]=out_path
+relevant_mount_point_arguments["--temporary-base-directory"]=out_path
+relevant_mount_point_arguments["--cache-directory"]=in_path
+relevant_mount_point_arguments["--save-directory"]=out_path
+relevant_mount_point_arguments["--task-dependencies-dot-file"]=out_file
+relevant_mount_point_arguments["--test-folder"]=in_path
+relevant_mount_point_arguments["--test-file"]=in_file
+
+function _get_mount_point_docker_arg() {
+  mount_point_path=
+
+}
+function _get_mount_point_argument() {
+  current_arg=$1
+  next_arg=$2
+  arg_type=$3
+  #  temp="${next_arg%\"}" NEED TO CHECK IF I NEED ARGUMENTS WITH/WITHOUT QUOTES
+  #  temp="${temp#\"}"
+  case $arg_type in
+  in_path)
+    if [[ -d $next_arg ]]; then
+      local dir_name=$(readlink -f $next_arg)
+      mount_point_arguments+="-v $dir_name:$dir_name:ro " #MOUNT READONLY!!!
+    else
+      echo "Input directory $next_arg for parameter $current_arg does not exist."
+      exit 1
+    fi
+    ;;
+  in_file)
+    if [[ -f $next_arg ]]; then
+      local rel_dir_name="$(dirname "${next_arg}")"
+      local dir_name=$(readlink -f rel_dir_name)
+      mount_point_arguments+="-v $dir_name:$dir_name:ro " #MOUNT READONLY!!!
+    else
+      echo "Input file $next_arg for parameter $current_arg does not exist."
+      exit 1
+    fi
+    ;;
+  out_path)
+    #Create out directories if necessary
+    if [[ ! -d $next_arg ]]; then
+      mkdir -p "$next_arg"
+    fi
+    local dir_name=$(readlink -f $next_arg)
+    mount_point_arguments+="-v $dir_name:$dir_name "
+    ;;
+  out_file)
+    local rel_dir_name="$(dirname "${next_arg}")"
+    #Create out directories if necessary
+    if [[ ! -d $next_arg ]]; then
+      mkdir -p "$rel_dir_name"
+    fi
+    local dir_name=$(readlink -f rel_dir_name)
+    mount_point_arguments+="-v $dir_name:$dir_name "
+    ;;
+  *)
+    echo "INVALID ARGUMENT. Please adjust variable relevant_mount_point_arguments in $0!"
+    exit 1
+    ;;
+  esac
+}
+
+function _get_mount_point_arguments() {
+  lenArgs="$#"
+  for ((idxArg = 1; idxArg < $lenArgs; idxArg++)); do
+    current_arg=${!idxArg}
+    next_arg_idx=$((idxArg + 1))
+    next_arg=${!next_arg_idx}
+    if [ -v relevant_mount_point_arguments[$current_arg] ]; then
+      _get_mount_point_argument $current_arg $next_arg ${relevant_mount_point_arguments[$current_arg]}
+    fi
+  done
+}
+
+mount_point_arguments=''
+_get_mount_point_arguments ${@}
+echo "MountPoint args:$mount_point_arguments"
+
 quoted_arguments=''
 for argument in "${@}"; do
   argument="${argument//\\/\\\\}"
   quoted_arguments="$quoted_arguments \"${argument//\"/\\\"}\""
 done
 
+###now collect list of mount points based on valid_mount_point_arguments....
+
 RUN_COMMAND="/script-languages-container-tool/starter_scripts/exaslct_without_poetry.sh $quoted_arguments; RETURN_CODE=\$?; chown -R $(id -u):$(id -g) .build_output &> /dev/null; exit \$RETURN_CODE"
```

Comment:
you need to add all the mounted path to the chown in the RUN_COMMAND. This also explains why we have files owned by root lingering on the system, after exaslct ran.